source/developers/concepts/hooks_extension_framework.rst

```diff
@@ -0,0 +1,136 @@
+|                    |  within a Django application.                                          |  to the caller or raises exceptions during                  |
+|                    |                                                                        |  processing.                                                |
++--------------------+------------------------------------------------------------------------+-------------------------------------------------------------+
+| **Use cases**      |  Send an email notification when a user enrolls in a course.           |  Prevent the enrollment of non-authorized users.            |
++--------------------+------------------------------------------------------------------------+-------------------------------------------------------------+
+
+How to know when to use an Event or a Filter?
+=============================================
+
+When to use an Open edX Event?
+------------------------------
+
+A developer might use an Open edX Event in order to perform the following actions. Note that this is not an exhaustive list.
+
+- Trigger custom logic or processing in response to specific actions within the platform, e.g., updating a search index after a course block is modified.
+- Communicate, synchronize, or coordinate with other components or services based on specific events or actions, e.g., send certificate data from LMS to credentials service to keep models up to date.
+- Integrate with external systems or services based on specific events or actions, e.g., send user data to third-party services upon registration for marketing purposes.
+
+.. For more detailed use cases, see the TODO: ADD REFERENCE TO REAL LIFE USE CASES documentation.
+
+In summary, events can be used to integrate application components with each other or with external services, allowing them to communicate, synchronize, and perform additional actions when specific triggers occur.
+
+You can review the :doc:`Open edX Events <openedx-events:index>` documentation for more information on :doc:`openedx-events:how-tos/using-events` in your project. This documentation includes a list of :doc:`openedx-events:reference/events` and much more.
+
+When to use an Open edX Filter?
+-------------------------------
+
+A developer might use an Open edX Filter in order to perform the following actions. Note that this is not an exhaustive list.
+
+- Enrich the data or parameters used to a specific component, e.g., fetch reusable LTI configurations from external plugins.
+- Enforce specific constraints or business rules of a specific component, e.g., don't allow registration for non-authorized email domains.
+- Implement additional features or behavior in a specific component, e.g., add registration extra fields to the user registration form.
+
+.. For more detailed use cases, see the TODO: ADD REFERENCE TO REAL LIFE USE CASES documentation.
```

Comment:
This is good to merge I guess after the real life examples docs are merged!

Reply:
@mariajgrimaldi before I merge, will you plan to address this in a different PR?

Reply:
I was going to, but I can quickly do it right now. I'll let you know once it's done.